asv_bench/benchmarks/pandas/series_methods.py

```diff
@@ -0,0 +1,278 @@
+
+    def time_clip(self, n):
+        self.s.clip(0, 1)
+
+
+class ClipDt:
+    def setup(self):
+        dr = date_range("20220101", periods=100_000, freq="s", tz="UTC")
+        self.clipper_dt = dr[0:1_000].repeat(100)
+        self.s = Series(dr)
+
+    def time_clip(self):
+        self.s.clip(upper=self.clipper_dt)
+
+
+class ValueCounts:
+
+    params = [[10**3, 10**4, 10**5], ["int", "uint", "float", "object"]]
+    param_names = ["N", "dtype"]
+
+    def setup(self, N, dtype):
+        self.s = Series(np.random.randint(0, N, size=10 * N)).astype(dtype)
+
+    def time_value_counts(self, N, dtype):
+        self.s.value_counts()
+
+
+class ValueCountsEA:
+
+    params = [[10**3, 10**4, 10**5], [True, False]]
+    param_names = ["N", "dropna"]
+
+    def setup(self, N, dropna):
+        self.s = Series(np.random.randint(0, N, size=10 * N), dtype="Int64")
+        self.s.loc[1] = NA
+
+    def time_value_counts(self, N, dropna):
+        self.s.value_counts(dropna=dropna)
+
+
+class ValueCountsObjectDropNAFalse:
+
+    params = [10**3, 10**4, 10**5]
+    param_names = ["N"]
+
+    def setup(self, N):
+        self.s = Series(np.random.randint(0, N, size=10 * N)).astype("object")
+
+    def time_value_counts(self, N):
+        self.s.value_counts(dropna=False)
+
+
+class Mode:
+
+    params = [[10**3, 10**4, 10**5], ["int", "uint", "float", "object"]]
+    param_names = ["N", "dtype"]
+
+    def setup(self, N, dtype):
+        self.s = Series(np.random.randint(0, N, size=10 * N)).astype(dtype)
+
+    def time_mode(self, N, dtype):
+        self.s.mode()
+
+
+class ModeObjectDropNAFalse:
+
+    params = [10**3, 10**4, 10**5]
+    param_names = ["N"]
+
+    def setup(self, N):
+        self.s = Series(np.random.randint(0, N, size=10 * N)).astype("object")
+
+    def time_mode(self, N):
+        self.s.mode(dropna=False)
+
+
+class Dir:
+    def setup(self):
+        self.s = Series(index=tm.makeStringIndex(10000))
+
+    def time_dir_strings(self):
+        dir(self.s)
+
+
+class SeriesGetattr:
+    # https://github.com/pandas-dev/pandas/issues/19764
+    def setup(self):
+        self.s = Series(1, index=date_range("2012-01-01", freq="s", periods=10**6))
+
+    def time_series_datetimeindex_repr(self):
+        getattr(self.s, "a", None)
+
+
+class All:
+
+    params = [[10**3, 10**6], ["fast", "slow"], ["bool", "boolean"]]
+    param_names = ["N", "case", "dtype"]
+
+    def setup(self, N, case, dtype):
+        val = case != "fast"
+        self.s = Series([val] * N, dtype=dtype)
+
+    def time_all(self, N, case, dtype):
+        self.s.all()
+
+
+class Any:
+
+    params = [[10**3, 10**6], ["fast", "slow"], ["bool", "boolean"]]
+    param_names = ["N", "case", "dtype"]
+
+    def setup(self, N, case, dtype):
+        val = case == "fast"
+        self.s = Series([val] * N, dtype=dtype)
+
+    def time_any(self, N, case, dtype):
+        self.s.any()
+
+
+class NanOps:
+
+    params = [
+        [
+            "var",
```

Comment:
We test some of these functions already in another place, right? Do we really need to test them again?

Reply:
cut these in half.